collections/iter.go

```diff
@@ -20,80 +20,85 @@ const (
 	OrderDescending Order = 1
 )
 
-// BoundInclusive creates a Bound of the provided key K
-// which is inclusive. Meaning, if it is used as Ranger.RangeValues start,
-// the provided key will be included if it exists in the Iterator range.
-func BoundInclusive[K any](key K) *Bound[K] {
-	return &Bound[K]{
-		value:     key,
-		inclusive: true,
-	}
+type keyModifierKind uint8
+
+const (
+	keyModifierKindNone keyModifierKind = iota
+	keyModifierKindNextKey
+	keyModifierKindPrefixEnd
+)
+
+// KeyModifier acts as an enum which defines different ways to encode a given key K to bytes.
+type KeyModifier[K any] struct {
```

Comment:
Does this need to be public?

I think `RangeKey` or `RangeBound` is a better name.

Reply:
It needs to be public to allow other people to implement their own API helpers of `Ranger` for their keys.